src/apps/main/database/collections/HashedSystemTreeCollection.ts

```
@@ -0,0 +1,121 @@
+import { DatabaseCollectionAdapter } from '../adapters/base';
+import { AppDataSource } from '../data-source';
+import { Repository } from 'typeorm';
+import * as Sentry from '@sentry/electron/main';
+import Logger from 'electron-log';
+import { ScannedItem } from '../entities/FileSystemHashed';
+
+export class HashedSystemTreeCollection implements DatabaseCollectionAdapter<ScannedItem> {
+  private repository: Repository<ScannedItem> = AppDataSource.getRepository('hashed_files');
+
+  async connect(): Promise<{ success: boolean }> {
+    return {
+      success: true,
+    };
+  }
+
+  async get(id: ScannedItem['id']) {
+    const match = await this.repository.findOneBy({ id });
+    return {
+      success: true,
+      result: match,
+    };
+  }
+
+  async getByPathName(pathName: ScannedItem['pathName']) {
+    const match = await this.repository.findOneBy({ pathName });
+    return {
+      success: true,
+      result: match,
+    };
+  }
+
+  async getAll() {
+    try {
+      const result = await this.repository.find();
+      return {
+        success: true,
+        result: result,
+      };
+    } catch (error) {
+      return {
+        success: false,
+        result: [],
```

The error is being lost here.